File: service-core/internal/modules/staff/domain/staff_profile.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type StaffProfile struct {
	ID          uuid.UUID
	UserID      uuid.UUID
	Name        string
	Username    string
	Phone       *string
	AvatarURL   *string
	LastLoginAt *time.Time
	CreatedAt   time.Time
	UpdatedAt   *time.Time
}

File: control-panel/src/models/StaffProfile.ts
export interface StaffProfile {
  staff_id: string;
  user_id: string;
  Name: string;
  Username: string;
  Phone?: string | null;
  AvatarURL?: string | null;
  LastLoginAt?: string | null;
  CreatedAt: string;
  UpdatedAt?: string | null;
}
